### PR TITLE
Fixes #8121 -- Report endpoint for data insight

### LINKED ISSUE #8121: Add analytics report endpoint

### PR DESCRIPTION
**Pull request by TeddyCr**
### Describe your changes :
Fixes #8121. Implements report endpoint to post refined analytics data fron webanalytics and entities

#
### Type of change :

- [ ] Bug fix
- [ ] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
Backend: @open-metadata/backend

### CODE REVIEW

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/AnalyticsReportResource.java [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+import org.openmetadata.service.util.RestUtil;
+import org.openmetadata.service.util.ResultList;
+
+@Slf4j
+@Path("/v1/analytics/report")
+@Api(value = "AnalyticsReport collection", tags = "AnalyticsReport collection")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Collection(name = "AnalyticsReport")
+public class AnalyticsReportResource extends EntityResource<ReportDefinition, AnalyticsReportRepository> {
```
Comment (nahuelverdugo) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/AnalyticsReportResource.java:55]: Should the name of these classes be aligned for consistency? e.g. `ReportDefinitionResource`, `ReportDefinition` and `ReportDefinitionRepository`?

File: bootstrap/sql/org.postgresql.Driver/v006__create_db_connection_info.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE IF NOT EXISTS analytics_report (
```
Comment (nahuelverdugo) [inline bootstrap/sql/org.postgresql.Driver/v006__create_db_connection_info.sql:1]: `analytics_report` or `report_definition`?

**Review by nahuelverdugo — DISMISSED** [review]
LGTM some minor comments of naming consistency

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/AnalyticsReportResource.java (hunk 1 above)
Comment (TeddyCr) [reply openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/AnalyticsReportResource.java:55]: The thing is that it collects report data as well. wdyt?

**Comment by TeddyCr** [comment]
@Sachin-chaurasiya for reference

**Comment by Sachin-chaurasiya** [comment]
> @Sachin-chaurasiya for reference

Thanks @TeddyCr

**Review by nahuelverdugo — DISMISSED** [review]

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/ReportDefinitionResource.java [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+import io.swagger.v3.oas.annotations.parameters.RequestBody;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import java.io.IOException;
+import java.util.List;
+import java.util.UUID;
+import javax.json.JsonPatch;
+import javax.validation.Valid;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.ws.rs.*;
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/ReportDefinitionResource.java:22]: avoid wild-card imports

**Review by harshach — DISMISSED** [review]
Left minor nits

File: openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/ReportDataResource.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+import org.openmetadata.schema.analytics.type.ReportData;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.jdbi3.ReportDataRepository;
+import org.openmetadata.service.resources.Collection;
+import org.openmetadata.service.security.Authorizer;
+import org.openmetadata.service.util.ResultList;
+
+@Slf4j
+@Path("/v1/analytic/reportData")
+@Api(value = "TestCase collection", tags = "TestCase collection")
```
Comment (nahuelverdugo) [inline openmetadata-service/src/main/java/org/openmetadata/service/resources/analytics/ReportDataResource.java:36]: ReportData? There are some references in the code to `TestCases`

File: openmetadata-spec/src/main/resources/json/schema/analytics/reportDataType/entityReportData.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
+{
+  "$id": "https://open-metadata.org/schema/tests/entityReportData.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "entityReportData",
+  "type": "object",
+  "javaType": "org.openmetadata.schema.analytics.type.EntityReportData",
```
Comment (nahuelverdugo) [inline openmetadata-spec/src/main/resources/json/schema/analytics/reportDataType/entityReportData.json:6]:
```suggestion
  "javaType": "org.openmetadata.schema.analytics.EntityReportData",
```

File: openmetadata-spec/src/main/resources/json/schema/analytics/reportData.json [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{
+  "$id": "https://open-metadata.org/schema/tests/reportData.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "reportData",
+  "type": "object",
+  "javaType": "org.openmetadata.schema.analytics.type.ReportData",
```
Comment (nahuelverdugo) [inline openmetadata-spec/src/main/resources/json/schema/analytics/reportData.json:6]:
```suggestion
  "javaType": "org.openmetadata.schema.analytics.ReportData",
```

File: openmetadata-spec/src/main/resources/json/schema/analytics/reportDataType/entityReportData.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
+{
+  "$id": "https://open-metadata.org/schema/tests/entityReportData.json",
```
Comment (nahuelverdugo) [inline openmetadata-spec/src/main/resources/json/schema/analytics/reportDataType/entityReportData.json:2]:
```suggestion
  "$id": "https://open-metadata.org/schema/analytics/entityReportData.json",
```

**Review by nahuelverdugo — APPROVED** [review]
LGTM I just miss some tests for the new resource.

**Pull request merged**